src/ophys_etl/transforms/convert_rois.py

```diff
@@ -0,0 +1,159 @@
+    pass
+
+
+class BinarizeAndCreateROIsInputSchema(ArgSchema):
+
+    suite2p_stat_path = Str(
+        required=True,
+        description=("Path to s2p output stat file containing ROIs generated "
+                     "during source extraction")
+    )
+
+    motion_corrected_video = Str(
+        required=True,
+        description=("Path to motion corrected video file *.h5")
+    )
+
+    motion_correction_values = InputFile(
+        required=True,
+        description=("Path to motion correction values for each frame "
+                     "stored in .csv format")
+    )
+
+    output_json = OutputFile(
+        required=True,
+        description=("Path to a file to write output data.")
+    )
+
+    maximum_motion_shift = Float(
+        default=30.0,
+        required=False,
+        allow_none=False,
+        description=("The maximum allowable motion shift for a frame before "
+                     "it is considered an anomaly and thrown out of "
+                     "processing")
+    )
+
+    abs_threshold = Float(
+        default=None,
+        required=False,
+        allow_none=True,
+        description=("The absolute threshold to binarize ROI masks against. "
+                     "If not provided will use quantile to generate "
+                     "threshold.")
+    )
+
+    binary_quantile = Float(
+        default=None,
+        required=False,
+        allow_none=True,
+        description=("The quantile against which an ROI is binarized. If not "
+                     "provided will use default function value of 0.1.")
+    )
+
+    @mm.post_load()
+    def check_args(self, data, **kwargs):
+        if data['binary_quantile'] is not None:
+            if data['binary_quantile'] <= 0:
+                raise BinarizeAndCreationException("Binary quantile must be"
+                                                   "positive value")
+        if data['abs_threshold'] is not None:
+            if data['abs_threshold'] <= 0:
+                raise BinarizeAndCreationException("Abs threshold must be "
+                                                   "positive value")
+        return data
+
+    @mm.post_load()
+    def check_file_existance(self, data, **kwargs):
+        """
+        have to check manually for existance of .h5 and .npy files because
+        InputFile fails on these, at least on windows because of a use of
+        generic open() to check for existance.
+        """
+        if not Path(data['suite2p_stat_path']).exists():
+            raise BinarizeAndCreationException("suite2p_stat_path supplied "
+                                               "is not a valid path.")
+        if not Path(data['motion_corrected_video']).exists():
+            raise BinarizeAndCreationException("motion_corrected_video "
+                                               "supplied is not a valid path.")
+        return data
+
+
+class BinarizeAndCreateROIsOutputSchema(ArgSchema):
+    old_rois = List(Dict(keys=Str),
```

Comment:
Does anyone know if I can specify a list or set of keys for agrschema to validate here instead of just type of key?

Reply:
So `TypedDict` basically. I don't think argschema has it. You can extend argschema by creating a new `field` and validating it.

Reply:
Argschema does import everything from `marshmallow` it seems. So `Schema` might work out of the box.

Reply:
I used another Schema class and defined it as nested and that resolves the issue! Thanks for the help.